src/ripple/nodestore/impl/DatabaseNodeImp.h

```diff
@@ -49,6 +49,7 @@ class DatabaseNodeImp : public Database
         , backend_(std::move(backend))
     {
         assert(backend_);
+        addToParent(parent);
```

Comment:
Doesn't this mean it's attached to two parents? One in the `Database::Stoppable()` (using `parent.getRoot()`) and one here using `parent`

Reply:
You could add a bool to verify that `addToParent` is only called once, as well as verify that the parent's root matches `m_root`